hw/ip/keymgr/doc/_index.md

```diff
@@ -178,56 +182,74 @@ An error code register is maintained {{< regref ERR_CODE >}} to check issues tha
-| Initialized    | Disable / Advance              | Advance to next Disabled state or CreatorRootKey.                                                                             |
-| Initialized    | Generate                       | Invalid operation error triggered with random data, output registers are updated.                                             |
-| CreatorRootKey | Disable / Advance / Generate   | Behave as expected.                                                                                                           |
-| OwnerIntKey    | Disable / Advance / Generate   | Behave as expected.                                                                                                           |
-| OwnerKey       | Disable / Advance / Generate   | Behave as expected.                                                                                                           |
-| Disabled       | Disable / Advance / Generate   | Invalid operation error triggered with random data, both working state and output registers are updated depending on command. |
+| Current State    | Legal Operations               |
+| -------------    | ------------------------------ |
+| Reset            | Advance                        |
+| Initialized      | Disable / Advance              |
+| CreatorRootKey   | Disable / Advance / Generate   |
+| OwnerIntKey      | Disable / Advance / Generate   |
+| OwnerRootKey     | Disable / Advance / Generate   |
+| Invalid/Disabled | None                           |
+
+*  All operations invoked during `Invalid` and `Disabled` states lead to invalid operation error.
+
+### Error Response
+In addition to alerts and interrupts, key manager may also update the working state key and relevant outputs based on current state.
+See the tables below for an enumeration.
+
+| Current State    | Invalid Command | Invalid Output | Invalid Input | Invalid Operation   |
+| -------------    | ----------------| -----------------------------------------------------|
+| Reset            | Not Possible    | Not Possible   | Not possible  | Not updated         |
+| Initialized      | Updated         | Updated        | Not updated   | Not updated         |
+| CreatorRootKey   | Updated         | Updated        | Not updated   | Not updated         |
+| OwnerIntKey      | Updated         | Updated        | Not updated   | Not updated         |
+| OwnerRootKey     | Updated         | Updated        | Not updated   | Not updated         |
+| Invalid/Disabled | Updated         | Updated        | Updated       | Updated             |
+
+*  During `Reset` state, the KMAC module is never invoked, thus certain errors are not possible.
+*  During `Initialized`, `CreatorRootKey`, `OwnerIntermediateKey` and `OwnerRootKey` states, a fault error causes the relevant key / outputs to be updated; however an operational error does not.
```

Comment:
not sure why not also update outputs if it's an Invalid Input error in `CreatorRootKey`, `OwnerIntermediateKey` and `OwnerRootKey`

Reply:
generate OP in Initialized is invalid operation, do you update output for it?

Reply:
because there are two categories of errors. There are fault errors and there are operational errors.
Operational means we might have done something wrong by accident via software.  So that one does nothing.  Fault error is like disable operation, so we always take the randomly computed values and update. 

Regarding invalid operation in initialized, please see table, it's not updated.

Reply:
btw, this is a change from before. The behavior before wasn't very consistent since i didn't really care what happened when someone made a generate call there.  But it's best to make it more consistent even if it doesn't matter.

Reply:
make sense, thanks for explanation.